### PR TITLE
ENH: `DBSCAN` SPMD iface

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
Let's merge #1400 first.
* `DBSCAN` SPMD iface
* example dbscan_spmd.py
based on #1400

### CODE REVIEW

**Comment by samir-nasibli** [comment]
GPU CI Job: http://intel-ci.intel.com/ee45c6f9-701f-f1b7-84a9-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU.

File: onedal/spmd/cluster/__init__.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
 if daal_check_version((2023, "P", 200)):
     from .kmeans import KMeans
 
-    __all__ = ["KMeans"]
```
Comment (samir-nasibli) [inline onedal/spmd/cluster/__init__.py:22]: Add dbscan import here

**Comment by samir-nasibli** [comment]
GPU CI Job: http://intel-ci.intel.com/ee45ea80-b281-f1c0-b78a-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU.

**Review by ethanglaser — APPROVED** [review]

**Pull request merged**